nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -0,0 +1,199 @@
+			LH.weight += 1 //Increment probabilty via SSEvent
+			if(LH.weight % 5 == 0)
+				message_admins("The ship has been out of combat for [last_combat_enter]. The weight of [LH.name] is now [LH.weight]")
+				log_game("The ship has been out of combat for [last_combat_enter]. The weight of [LH.name] is now [LH.weight]")
+		if(istype(BR))
+			BR.weight += 1 //Increment probabilty via SSEvent
+			if(BR.weight % 5 == 0)
+				message_admins("The ship has been out of combat for [last_combat_enter]. The weight of [BR.name] is now [BR.weight]")
+				log_game("The ship has been out of combat for [last_combat_enter]. The weight of [BR.name] is now [BR.weight]")
+
+/datum/controller/subsystem/starsystem/proc/event_info() //Admin command for debugging output
+	var/datum/round_event_control/lone_hunter/LH = locate(/datum/round_event_control/lone_hunter) in SSevents.control
+	var/datum/round_event_control/belt_rats/BR = locate(/datum/round_event_control/belt_rats) in SSevents.control
+	message_admins("Current time: [world.time] | Last Combat: [last_combat_enter] | Modifier: [modifier] | [LH.name]: [LH.weight] | [BR.name]: [BR.weight]")
+
+/datum/controller/subsystem/starsystem/proc/weighting_reset() //All overmap combat events need to be populated in this proc
+	var/datum/round_event_control/lone_hunter/LH = locate(/datum/round_event_control/lone_hunter) in SSevents.control
+	var/datum/round_event_control/belt_rats/BR = locate(/datum/round_event_control/belt_rats) in SSevents.control
+	LH.weight = 0
+	BR.weight = 0
+
+/datum/controller/subsystem/starsystem/New()
+	. = ..()
+	instantiate_systems()
+
+/datum/controller/subsystem/starsystem/proc/instantiate_systems()
+	cycle_gameplay_loop() //Start the gameplay loop
+	cycle_bounty_timer() //Start the bounty timers
+	var/datum/starsystem/SS = new
+	hyperspace = SS //First system defaults to "hyperspace" AKA transit.
+	for(var/instance in subtypesof(/datum/starsystem))
+		var/datum/starsystem/S = new instance
+		systems += S
+
+///////SPAWN SYSTEM///////
+
+/datum/controller/subsystem/starsystem/proc/find_main_overmap()
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.main_overmap)
+			return OM
+
+/datum/controller/subsystem/starsystem/proc/find_main_miner()
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.main_miner)
+			return OM
+
+/datum/controller/subsystem/starsystem/proc/find_system(obj/structure/overmap/OM) //Used to determine what system a ship is currently in. Famously used to determine the starter system that you've put the ship in.
+	var/datum/starsystem/found
+	for(var/datum/starsystem/S in systems)
+		for(var/thez in SSmapping.levels_by_trait(S.level_trait))
+			if(thez == OM.z)
+				found = S
+				break
+	return found
+
+/datum/controller/subsystem/starsystem/proc/modular_spawn_enemies(obj/structure/overmap/OM, datum/starsystem/target_sys)//Select Ship to Spawn and Location via Z-Trait
+	var/list/levels = SSmapping.levels_by_trait(target_sys.level_trait)

```

Comment:
```suggestion
	var/datum/zlevel/level = SSmapping.levels_by_trait(target_sys.level_trait)[1] //[1] because arrays start at 1 in byond because reasons.

```
Then modify the "levels.len" stuff accordingly

Reply:
We'd do this to remove all the edge cases of if(levels.len == 1.. -> 2, etc)

Reply:
More than one zlevel acting as a specific system is actually an error state that's good to know about imo. And you'd have to handle the empty list case regardless of what you do.